packages/website/blog/2022-09-03-automated-rule-docs-with-docusaurus-and-remark.md

```diff
@@ -0,0 +1,228 @@
+  children: [
+    {
+      children: meta.docs.description
+        .split(/`(.+?)`/)
+        .map((value, index, array) => ({
+          type: index % 2 === 0 ? 'text' : 'inlineCode',
+          value: index === array.length - 1 ? `${value}.` : value,
+        })),
+      type: 'paragraph',
+    },
+  ],
+  type: 'blockquote',
+});
+```
+
+For example, [typescript-eslint.io/rules/array-type](https://typescript-eslint.io/rules/array-type) has the equivalent of this blockquote added now:
+
+```md
+> Require using either `T[]` or `Array<T>` for arrays.
+```
+
+Quite a few other modifications to the AST involve adding AST nodes that are expected to exist on the page.
+It ensures expected headings such as _`## How to Use`_ and _`## Options`_ exist, inserting them if not.
+
+### Adding a JSX AST Node
+
+Remember when the concept of MDX files including JSX tags was mentioned earlier in this blog post, and then never referenced again?
+We did end up injecting one React component as a JSX tag in the plugin.
+
+In our [MDX component configuration](https://github.com/typescript-eslint/typescript-eslint/blob/39829c01906f326fec94e9b3a5fdb1730eb02002/packages/website/src/theme/MDXComponents/index.tsx#L2), `'rule-attributes'` maps to a [`RulesAttribute` React component](https://github.com/typescript-eslint/typescript-eslint/blob/39829c01906f326fec94e9b3a5fdb1730eb02002/packages/website/src/theme/MDXComponents/RuleAttributes.tsx#L7) that renders a list of rule metadata.
+That list is a little too complex to easily render directly in Markdown.
+
+MDX builds on Remark and additionally supports nodes of type `'jsx'` that include raw JSX syntax:
+
+```ts
+const attributesH2Index = root.children.findIndex(
+  child => nodeIsHeading(child) && child.depth === 2,
+);
+
+root.children.splice(attributesH2Index, 0, {
+  type: 'jsx',
+  value: `<rule-attributes name="${file.stem}" />`,
+});
+
+function nodeIsHeading(node: unist.Node): node is mdast.Heading {
+  return node.type === 'heading';
+}
+```
+
+## What's Next
+
+We're relieved that we no longer have to manually update scores of documentation on every code change.
+This plugin's automation frees up our documentation time to focus on improving the contents of the pages themselves.
+You can query our [issue tracker](https://github.com/typescript-eslint/typescript-eslint/issues) for [issues with the `documentation` label](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3Adocumentation) to see what's coming up next.
+I'm excited to focus in particular on [Docs: Proofread rule docs for clarity (#4861)](https://github.com/typescript-eslint/typescript-eslint/issues/4861).
+
+## Appreciation and Thanks
+
+We'd like to extend thanks to [Joshua Cena](https://github.com/Josh-Cena), one of the Docusaurus maintainers who also has been helping us with Docusaurus — and helped proofread [this blog post's PR](https://github.com/typescript-eslint/typescript-eslint/pull/5593).
```

Comment:
```suggestion
We'd like to extend thanks to [Joshua Chen](https://github.com/Josh-Cena), one of the Docusaurus maintainers who also has been helping us with Docusaurus — and helped proofread [this blog post's PR](https://github.com/typescript-eslint/typescript-eslint/pull/5593).
```

(Also apart from the obviously wrong "Josh Cena" I have no preference of Josh or Joshua, just FYI)

Reply:
Got it, will keep to this from now on!